langchain/output_parsers/json.py

```diff
@@ -9,14 +9,21 @@
 
 def parse_json_markdown(json_string: str) -> dict:
     # Try to find JSON string within triple backticks
-    match = re.search(r"```(json)?(.*?)```", json_string, re.DOTALL)
+
+    # The amount of backticks at the end of the LLM response can vary,
+    # so we use a regex to match what we want even if we have fewer than 3 backticks
+    # As a tradeoff, we might fetch some backticks in the command below: match.group(2)
+    # We will get rid of them using rstrip() later
+    match = re.search(r"```(json)?(.*)(`{1,3})?", json_string, re.DOTALL)
 
     # If no match found, assume the entire string is a JSON string
     if match is None:
         json_str = json_string
     else:
         # If match found, use the content within the backticks
         json_str = match.group(2)
+        # In case we have picked some trailing backticks, get rid of them
+        json_str = json_str.rstrip("`")
```

Comment:
How about ````json_str.strip().rstrip("`")````?

Currently, the parsing fails on strings with whitespace trailing the backticks, since it is trying to strip backticks from the right side of the string and hits whitespace first.

Reply:
Totally agree, never thought of it!